### PR TITLE
Added Information to space-complexity.md

### LINKED ISSUE #59: new lesson: space-complexity
### ⭐ Request Type

I want to add information in space-complexity.md
Structure:
- Concept
- Need & Application
- Calculation of space complexity
- Explanation about calculation using examples
- Space complexities of common algorithms

### 💻 Location

Location is at

section: 7
lesson: C
order: 7C
filename: space-complexity.md

### ❌ Additional Info

_No response_

### PR DESCRIPTION
**Pull request by ParnaviKulkarni**
Added information to space-complexity.md.
fixes #59 
#### Description of work:
- Explanation of concept & need
- Process of calculation
- Explanation of process with examples
- Standard space complexities of common algorithms

#### References used : 
- Sartaj & Sahani DSA book
- geeksforgeeks
- stackoverflow
- baeldong.com
- studytonight.com

### CODE REVIEW

File: lessons/space-complexity.md [hunk 1]
```diff
@@ -5,3 +5,66 @@ order: "7C"
...
+
+Now that we have learned about 'time' aspect of performance analysis of an algorithm, let's move on to memory aspect of the same.
+
+***Space complexity of an algorithm is basically the amount of memory it needs to run to completion.***
+
+It is a way to establish approximate relationship between size of input data and primary memory used by the algorithm to produce the expected result.
+
+Calculation of space complexity used to hold much more significance in early days of computing than it does now. This is because most machines today have large memories and the user does not need to worry about running out of memory for running a program or two. But it is a crucial estimate where the physical memory is limited or closely monitored.
+
+### **Calculation of Space Complexity**
```
Comment (Utkarsh1504) [inline lessons/space-complexity.md:17]:
```suggestion
## **Calculation of Space Complexity**
```
please make this heading from `h3` to **`h2`**

File: lessons/space-complexity.md [hunk 2]
```diff
@@ -5,3 +5,66 @@ order: "7C"
...
+  if (n<=0) return 1;
+  else  return n * (n - 1);
+}
+```
+In this case there are 3 statements (1 if & 2 return statements). The depth of recursion is *n + 1*. Thus the recursion stack space needed is >=3(n+1). So we can say, space complexity is O(n) i.e. linear.
+
+***NOTE: Space complexity might differ from machine to machine based on the programming language, the compiler etc.***
+
+
+### **Space Complexities of Common Algorithms**
```
Comment (Utkarsh1504) [inline lessons/space-complexity.md:54]:
```suggestion
## **Space Complexities of Common Algorithms**
```
make `h2` here also

**Review by Utkarsh1504 — CHANGES_REQUESTED** [review]
Rest all the thing looks good to me just change the heading from h3 to h2 thank you!

**Comment by ParnaviKulkarni** [comment]
I've made the changes. Will you please review it once again?

**Review by Utkarsh1504 — APPROVED** [review]
looks good to me.

**Comment by Utkarsh1504** [comment]
Thank you for your contribution! 🎈🎉✨

**Comment by ParnaviKulkarni** [comment]
Thank you so much @Utkarsh1504 And also thank you for guiding me throughout this process and giving me chance to contribute even when I faced issues.

**Pull request merged**